### PR TITLE
Update webpack 5

### PR DESCRIPTION
**Pull request by diclophis**
This should add support for webpack 5: see https://webpack.js.org/api/compiler-hooks/ and https://medium.com/@sheng_di/webpack-4-migration-notes-65f2f4b79b8f#767f

### CODE REVIEW

**Comment by yura999999999** [comment]
@diclophis is it working?

In manual https://webpack.js.org/contribute/plugin-patterns/#monitoring-the-watch-graphx

Since webpack 5, compilation.fileDependencies, compilation.contextDependencies and compilation.missingDependencies are now a Set

```
TypeError: Cannot read property 'push' of undefined
    at applyAfterEmit (/node_modules/unused-files-webpack-plugin/lib/index__8.0.0__.js:63:16)
```

**Comment by grzegorzjudas** [comment]
Is there some more work needed on this? Or could this be merged?

**Comment by diclophis** [comment]
@grzegorzjudas I think this is ready for merge ... I have only run it through 1 application for testing tho

**Comment by yura999999999** [comment]
@grzegorzjudas Hi, first of all thanks for the pull request

But i have a few questions:
1. Why are there so many line and formatting changes? No code changes
2. Why is the afterEmit hook replaced with done hook? Those who have used this plugin before may have problems with this.

**Comment by grzegorzjudas** [comment]
@yura999999999 I am not the author - @diclophis is :)

Regarding the formatting, I was wondering about it too. Perhaps @diclophis can elaborate, but for one, it's difficult to review it because of that.

**Comment by yura999999999** [comment]
@grzegorzjudas yes, of course, I wrote to @diclophis )) sorry

**Comment by diclophis** [comment]
I think the crux of the patch is this:

image

I cannot attest to the compatibility of these changes as far as other projects go ... the internal test suite is green though... 

My system must have some other babel generator settings that are not matched with the original project, most of the files in lib that were changed I think are "compiled" files ... I won't have much more time to look into this issue, we are currently looking at removing this package from our dep tree.

**Comment by mrbalov** [comment]
Is there a plan to merge it?

File: package.json [hunk 1]
```diff
@@ -74,6 +74,6 @@
     "prettier": "^1.11.1",
     "rimraf": "^2.6.2",
     "standard-version": "^4.3.0",
-    "webpack": "^3.8.1"
+    "webpack": "^5"
```
Comment (mrbalov) [inline package.json:77]: 👍🏾

**Review by mrbalov — APPROVED** [review]

**Comment by JohnSpindler** [comment]
👋🏼 checking in on the status of this PR. I'd love to be able to add this into my webpack build again 😄

**Comment by zoexx** [comment]
👀 still waiting for updates

**Comment by diclophis** [comment]
Please re-open and merge when needed.

**Comment by isrmicha** [comment]
reopen

**Comment by pepijnolivier** [comment]
@tomchentw Could you please review and hopefully merge this ? 🙏

**Comment by chekrd** [comment]
Our project moved from unused-files-webpack-plugin to [webpack-deadcode-plugin](https://github.com/MQuy/webpack-deadcode-plugin) and I can recommend you to do the same in the context of Webpack@5. This repo seems to be dead.

**Pull request closed without merging**